### ISSUE TITLE
Getting message "This app may have issues in the web version of teams" when adding the app

### ISSUE THREAD

**Issue by nbelyh** [OfficeDev/microsoft-teams-library-js#2185]
Started getting the issue when adding the app in a WEB browser:

The app is a few years old (not a new one).

![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/528366/fca5efd9-ba4b-43d8-9800-1f1b55a477b3)

If proceeding, there is an issue with login, this one (it launches the prompt in a complete new window, and getting teams token fails).
![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/528366/02759537-eec4-4e05-a9fe-7ec43d082d9e)

"Invalid authResponse received from msla for resource api ...."
The application is calling `microsoftTeams.authentication.getAuthToken()` at this point to get the teams token.

Any ideas where to look for? What does the first message mean, what re the changes?
If I switch to the "classic" (old) teams in the web browser, all works as expected.
Also in the desktop teams client it works like always.

**Comment by Prasad-MSFT** [comment]
@nbelyh - The error message indicates that there is an issue with the authentication process using MSAL (Microsoft Authentication Library) in your app. Please ensure that your authentication flow is compatible with both the desktop and web versions of Teams. and check if there have been any changes in the authentication process over the years that may affect the functionality of your app. Check if there are any outdated dependencies or libraries related to authentication that need to be updated to ensure compatibility with the latest Teams SDK and authentication mechanisms.

Also, could you please provide the app package if possible so that we can repro this at our end as well?

**Comment by Prasad-MSFT** [comment]
Hi @nbelyh - We tried to repro this using **VisPlan** application. The first time when we tried to add the app in any team, it showed us the same error as you are encountering in **New Teams** **Web browser**. However, the next time when we tried to do the same it asked us to login and give consent using credentials and it worked successfully.

**Comment by nbelyh** [comment]
@Prasad-MSFT 
Yes, exactly this is the problemn. I.e. this happens only FIRST TIME, when user did not consent yet.

**Comment by Prasad-MSFT** [comment]
@nbelyh - I agree with that. For new users that will not be good experience. We have raised a bug for the same. We will inform you once we get any further update from engineering team. 

Thanks!

**Comment by rahul-chanana** [comment]
We are facing the same issue with our app. Please provide an update about status of this bug.

**Comment by jke-r** [comment]
@Prasad-MSFT : We have the same problem. This happens with our well established app as well with our newly developed app.
Everything works fine in Desktop app. But in the browser we get the same screen, everytime(!) the app is loaded. This is a huge issue as this will give our user a very trust unworthy experience.

We have recently updated to teams-js 2.21.0. But it worked stable after the migration for several weeks. We did not change anything else so far.

Interesting thing: This also happens without token retreival. Our latest app (currently in development) does not receive any tokens from teams-js. Only the context information.

Our environment:
* Windows 10, Edge browser
* Manifest version 1.16
* So is no error message in console log while the screen is shown or after.

Do you have any update on that matter?

**Comment by Prasad-MSFT** [comment]
We will check this with engineering team and will update this thread once we hear from them.
Thanks!

**Comment by jke-r** [comment]
@Prasad-MSFT : Any update on this? It still happens and is really annoying for our users and us.

**Comment by jke-r** [comment]
@Prasad-MSFT, @nbelyh :

We have just checked:
* Microsoft Edge: Teams (V2): When we open our app, this happens everytime(!).
* Google Chrome: Teams (V2): When we open our app, it works as expected. User is automatically logged in with SSO.

**Comment by Trkrkrl** [comment]
Hi  I was getting similar error with same message as @nbelyh .
In Azure  , Microsoft Entra we add client apps ID so they can access. 
These ıd's mentioned here: https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-register-aad
After adding these IDs:
1fec8e78-bce4-4aaf-ab1b-5451cc387264 =>for Teams mobile or desktop application

5e3ce6c0-2b1f-4285-8d4b-75ee78787346 =>for Teams web application
![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/94923090/3bf092e7-2497-480e-8f71-2cccf2dce28f)


my sso started working.
I hope it helps

**Comment by jker-org** [comment]
@Trkrkrl : Thank you for the hint. Unfortunately, that is not the cause for that problem. Those IDs have been added years ago. As @nbelyh mentioned: This is a very specific problem that only occurs with Teams V2 on the web.

@Prasad-MSFT : Any update from you on this? This is really frustrating for our users and leaves a bad impression about our app.

**Comment by jbmyid** [comment]
The warning page is shown when the browser has third-party cookies disabled eg: chrome incognito or safari.
I wonder if we can disable this warning with the app manifest parameter.

**Comment by ahukkanen** [comment]
We are running an embedded Teams "Tab application" and also saw this message. Following above what @jbmyid said, I have gone through the process of allowing 3rd party cookies for the Microsoft services following this guidance:
https://learn.microsoft.com/en-us/microsoftteams/troubleshoot/teams-sign-in/sign-in-loop

After that, the error is gone.

Would be of course nice if we could get rid of this error automatically for all users, even with 3rd party cookies disabled.

**Comment by jker-org** [comment]
@jbmyid , @ahukkanen : Thanks for your input. Unfortunately, it does not fix the situation for our apps (and others). We usually test with Edge, here the error occurs. However, third-party cookies are allowed.

![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/130495478/e8da9974-d13f-4149-aecf-430921413a07)

**Comment by Draginfable** [comment]
Same thing here. I was testing with third-party cookies disabled to implement the Storage Access API, after being added. (https://github.com/OfficeDev/microsoft-teams-library-js/issues/1460)

Even though the Storage Access API works fine and the cookies of our app work fine when third-party cookies disabled, MS Teams shows this misleading warning.

cc @Prasad-MSFT I agree with @jbmyid that there should be a manifest flag to disable this warning.

**Comment by jker-org** [comment]
Hi @Prasad-MSFT, is there any update to this bug?

**Comment by tiennguyen1293** [comment]
I'm facing this issue on my App. It happened recently.

![image](https://github.com/user-attachments/assets/865bab1a-b709-4980-857b-c25f1477881e)

**Comment by egm0121** [comment]
> Even though the Storage Access API works fine and the cookies of our app work fine when third-party cookies disabled, MS Teams shows this misleading warning.

Hi @ChetanSharma-msft and  @Prasad-MSFT what is the latest regarding this warning? 
Is there a manifest flag to disable this warning?

**Comment by ChetanSharma-msft** [comment]
Hello @egm0121 - Thanks for following up on this. We’ve raised the bug and are actively checking with the engineering team for updates. We’ll keep you informed as soon as there’s any progress or guidance on this matter.

**Comment by AnandDMore** [comment]
I have found that this warning appears when third-party cookies are disabled. It works fine in the desktop application, but for the web version, third-party cookies must be enabled if they are disabled. In Incognito mode, they are disabled by default.

**Comment by egm0121** [comment]
Hi @ChetanSharma-msft  and @Prasad-MSFT is there any update regarding how we can disable the warning: "This application may have issues in the web version of Teams".
We are making use of the Storage Access API to enable our authentication flow even if third-party cookies are blocked by default (eg. Safari, Chrome Incognito window).
The warning is misleading for our users and we currently have no way to disable it.

**Comment by adamreisnz** [comment]
We too are running into this issue, @Prasad-MSFT , would it be possible to disable this warning, as it is indeed misleading and has nothing to do with our app as we don't rely on cookie based authentication.

**Comment by rsavioli** [comment]
Hi @ChetanSharma-msft @Prasad-MSFT any update?

**Comment by sayali-MSFT** [comment]
Thank you for your inquiry regarding the status of the bug. We are currently checking in with the engineering team for any updates. 

We appreciate your patience and will inform you as soon as we have more information. Thank you!

**Comment by egm0121** [comment]
@Prasad-MSFT @sayali-MSFT @ChetanSharma-msft is there a plan to allow team apps to programmatically hide this full-page warning?
